### PR TITLE
Add support for CoreHaptics on iOS devices

### PR DESCRIPTION
**Pull request by Leicas**
Add more complex vibrations for iOS devices, the same pattern of various amplitude can now be rendered on both platforms.

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -32,13 +32,27 @@ class VibratingApp extends StatelessWidget {
                     onPressed: () {
                       final snackBar = SnackBar(
                         content: Text(
-                            'Pattern: wait 0.5s, vibrate 1s, wait 0.5s, vibrate 2s, wait 0.5s, vibrate 3s, wait 0.5s, vibrate 0.5s'),
+                            'Pattern: wait 0.5s, vibrate 1s, wait 1s, vibrate 2s, wait 0.5s, vibrate 3s, wait 0.5s, vibrate 0.5s'),
```
Comment (benjamindean) [inline example/lib/main.dart:35]: Should still be 0.5, right?

File: ios/Classes/SwiftVibrationPlugin.swift [hunk 2]
```diff
@@ -1,29 +1,171 @@
 import Flutter
 import UIKit
 import AudioToolbox
+import CoreHaptics
 
+@available(iOS 13.0, *)
```
Comment (benjamindean) [inline ios/Classes/SwiftVibrationPlugin.swift:6]: Could annotation the entire class break something?

File: ios/Classes/SwiftVibrationPlugin.swift [hunk 3]
```diff
@@ -1,29 +1,171 @@
...
+                hapticEvents.append(e)
+                
+                // Add duration to relative time
+                rel += duration
+            }
+            
+            // Try to play engine
+            do {
+                if let engine = SwiftVibrationPlugin.engine {
+                    let patterntoplay = try CHHapticPattern(events: hapticEvents, parameters: [])
```
Comment (benjamindean) [inline ios/Classes/SwiftVibrationPlugin.swift:154]: `patternToPlay`?

**Comment by benjamindean** [comment]
Thank you! I will test it out as soon as I can.

File: example/lib/main.dart (hunk 1 above)
Comment (Leicas) [reply example/lib/main.dart:35]: Yes, right !

File: ios/Classes/SwiftVibrationPlugin.swift (hunk 2 above)
Comment (Leicas) [reply ios/Classes/SwiftVibrationPlugin.swift:6]: I honestly don’t know.

File: ios/Classes/SwiftVibrationPlugin.swift (hunk 3 above)
Comment (Leicas) [reply ios/Classes/SwiftVibrationPlugin.swift:154]: Yes.

**Pull request merged**